lib/cache/cache.go

```diff
@@ -1259,7 +1259,7 @@ func (c *Cache) fetch(ctx context.Context) (fn applyFn, err error) {
 
 			applyfn, err := collection.fetch(ctx)
 			if err != nil {
-				return trace.Wrap(err)
+				return trace.Wrap(err, fmt.Sprintf("failed to fetch resource: %q", kind))
```

Comment:
I think `trace.Wrap` support fmt-like expressions
 
```suggestion
				return trace.Wrap(err, "failed to fetch resource: %q", kind)
```